crates/client/settlement_client/src/starknet/mod.rs

```diff
@@ -0,0 +1,470 @@
+            &account,
+            deployed_address,
+            StateUpdate { block_number: 100, global_root: data_felt, block_hash: data_felt },
+        )
+        .await?;
+        sleep(Duration::from_secs(5)).await;
+
+        let last_verified_block_hash = starknet_client.get_last_verified_block_hash().await?;
+        assert_eq!(last_verified_block_hash, data_felt, "Block hash should match");
+
+        Ok(())
+    }
+
+    #[serial]
+    #[tokio::test]
+    async fn get_last_state_root_works_starknet_client() -> anyhow::Result<()> {
+        // Here we need to have madara variable otherwise it will
+        // get dropped and will kill the madara.
+        #[allow(unused_variables)]
+        let (account, deployed_address, madara) = prepare_starknet_client_test().await?;
+        let l1_block_metrics = L1BlockMetrics::register()?;
+        let starknet_client = StarknetClient::new(StarknetClientConfig {
+            url: Url::parse(format!("http://127.0.0.1:{}", MADARA_PORT).as_str())?,
+            l2_contract_address: deployed_address,
+            l1_block_metrics,
+        })
+        .await?;
+
+        // sending state updates :
+        let data_felt = Felt::from_hex("0xdeadbeef")?;
+        send_state_update(
+            &account,
+            deployed_address,
+            StateUpdate { block_number: 100, global_root: data_felt, block_hash: data_felt },
+        )
+        .await?;
+        sleep(Duration::from_secs(5)).await;
+
+        let last_verified_state_root = starknet_client.get_last_state_root().await?;
+        assert_eq!(last_verified_state_root, data_felt, "Last state root should match");
+
+        Ok(())
+    }
+
+    #[serial]
+    #[tokio::test]
+    async fn get_last_verified_block_number_works_starknet_client() -> anyhow::Result<()> {
+        // Here we need to have madara variable otherwise it will
+        // get dropped and will kill the madara.
+        #[allow(unused_variables)]
+        let (account, deployed_address, madara) = prepare_starknet_client_test().await?;
+        let l1_block_metrics = L1BlockMetrics::register()?;
+        let starknet_client = StarknetClient::new(StarknetClientConfig {
+            url: Url::parse(format!("http://127.0.0.1:{}", MADARA_PORT).as_str())?,
+            l2_contract_address: deployed_address,
+            l1_block_metrics,
+        })
+        .await?;
+
+        // sending state updates :
+        let data_felt = Felt::from_hex("0xdeadbeef")?;
+        let block_number = 100;
+        send_state_update(
+            &account,
+            deployed_address,
+            StateUpdate { block_number, global_root: data_felt, block_hash: data_felt },
+        )
+        .await?;
+        sleep(Duration::from_secs(5)).await;
+
+        let last_verified_block_number = starknet_client.get_last_verified_block_number().await?;
+        assert_eq!(last_verified_block_number, block_number, "Last verified block should match");
+
+        Ok(())
+    }
+}
```

Comment:
can we test `listen_for_update_state_events`

Reply:
oh ok, I noticed we've a test for this in the common starknet/eth file. can we move it inside eth and starknet because they've different logics?

Reply:
But shouldn't the tests be with the module ?